pkg/virt-config/virt-config.go

```diff
@@ -63,7 +63,7 @@ const (
 	DefaultARCHOVMFPath                             = "/usr/share/OVMF"
 	DefaultAARCH64OVMFPath                          = "/usr/share/AAVMF"
 	DefaultMemBalloonStatsPeriod             uint32 = 10
-	DefaultCPUAllocationRatio                       = 10
+	DefaultCPUAllocationRatio                       = 1
```

Comment:
In our understanding, if a user has configured the `cpuAllocationRatio`, he must want to enable CPU overcommit for the virtual machine.
So whether set `vmi.spec.domain.cpu` or `vmi.spec.resource.request/limit.cpu`, as long as the configuration `cpuAllocationRatio` in kubevirt cr, that must want the CPU to overcommit. If not, remove the `cpuAllocationRatio` configuration from the Kubevirt CR.
According to the above conditions, then DefaultCPUAllocationRatio == 1, when calculating the CPU of virt-launcher pod, as long as more than 1, some one must have set the `cpuAllocationRatio`.

Reply:
Well, we should always overcommit CPU by default to increase the density of pods/VMIs. This is also the default behavior in Kubernetes in general. Each pod by default gets 1/10 of a cputime of a core. If one full core is considered as 1000 mil, the default per pod is 100mil. 
KubeVirt's DefaultCPUAllocationRatio follows k8s and allocated 1/10 of a core for a given pod.

On the other hand, KubeVirt must scale this CPU request with the number of vCPUs the VMI guest requires. 
Without scaling, as the number of vCPUs increases, each will get a lower CPU time when competing with other processes on the node or even with VMIs with a lower amount of vCPUs.
The `cpuAllocationRatio` merely normalizes the requested compute container CPUs based on the total number of vCPUs * cpuAllocationRatio

For example, using the default cpuAllocationRatio and 6 vCPUs, the VMI will look like 

```
spec:
  domain:
    cpu:
      sockets: 1
      cores: 6
      threads: 1
```
The vmipod will request 600mil of cpu time.
```
    resources:
      requests:
        cpu: 600m
```

and 

```
spec:
  domain:
    cpu:
      sockets: 1
      cores: 1
      threads: 1
```
The vmipod will get the default 100mil of cpu time.
```
    resources:
      requests:
        cpu: 100m
```

VMI users can also request guaranteed CPUs and allocate the exact amount of vCPU from the host.
In that case, there will be no overcommit. The vmi pod will be assigned a Guaranteed QoS and will get exclusive CPUs.
This can simply be achieved by requesting a dedicatedCpuPlacement. [1]

```
spec:
  domain:
    cpu:
      sockets: 2
      cores: 1
      threads: 1
      dedicatedCpuPlacement: true
```

If you are simply looking to change the `cpuAllocationRatio`, you can do so by editing the kubevirt CR instead of changing the default.


[1] https://kubevirt.io/user-guide/virtual_machines/dedicated_cpu_resources/#dedicated-cpu-resources